### PR TITLE
Added pre-commit hook script to help teams avoid control their data

### PR DESCRIPTION
**Pull request by rmbielby**
We need a method to prevent unpublished data being published to Team's github repositories. We've put together a pre-commit hook such that any files that might be expected to contain some form of data are required to be recorded in a log file. For each file, the logfile requires the full path to that file and a flag listing whether the file contains unpublished, published or reference data. 

Prior to allowing any commit, the pre-commit hook searches for any .csv, .xlsx, .dat, .tex or .pdf files. If any of the files it finds are not present in the log file, the commit is aborted with the following messaging:

image

If any of these files are present in the logfile, but are not listed as published or reference and are not also present in .gitignore, then the commit will also be aborted:

image

### CODE REVIEW

**Review by cjrace — CHANGES_REQUESTED** [review]
The pre-commit hooks don't seem to be working - have committed files that aren't in the log, as well as made the example file unpublished and committed that?

**Review by chfoster — APPROVED** [review]
Tested last week & couldn't break it - seemed to work as it should!

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 .Rhistory
 .RData
 .Ruserdata
+*unpublished*
```
Comment (cjrace) [inline .gitignore:5]: Does this automatically pick up unpublished files or are we expecting teams to add them here (the pre-commit hook message suggests the latter). If we're expecting analysts to add files can we stick in an example file for them to see here?

**Review by cjrace — CHANGES_REQUESTED** [review]
Had a look, all working well though only seems to work on R 4.2 or greater, I was on R 4.1.3 when I tried at first and it wouldn't install all the dependencies properly.

Just the one thing around adding an example file in the gitignore for analysts and then I think it's good to merge

File: .gitignore (hunk 1 above)
Comment (chfoster) [reply .gitignore:5]: Could add a comment instead to explain?

Comment (rmbielby) [reply .gitignore:5]: @cjrace Have now added an example file that gets excluded from commits, but allows the commit to go ahead. i.e. it's listed in [datafiles_log.csv](https://github.com/dfe-analytical-services/shiny-template/blob/020cb4a6d2a5ab978a7dae4ca88e420eff4a663a/datafiles_log.csv] as unpublished. 

[some_unpublished_data.csv](https://github.com/dfe-analytical-services/shiny-template/blob/020cb4a6d2a5ab978a7dae4ca88e420eff4a663a/data/some_unpublished_data.csv)

**Review by cjrace — APPROVED** [review]

**Pull request merged**